File: backend/DimiTours/Models/Question.cs
using DimiTours.Enum;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DimiTours.Models
{
    public class Question
    {
        public int Id { get; set; }

        [Required]
        public QuestionType Type { get; set; }

        [Required]
        [MaxLength(255)]
        public string Text { get; set; }

        [Required]
        public List<Answer> Answers { get; set; }
        public List<ActivityQuestion> ActivityQuestions { get; set; }


    }
}

File: frontend/src/models/Question.ts
import {Answer} from "./Answer.ts";

export enum QuestionType {
    HISTORY,
    GEOGRAPHY,
    MYTHOLOGY,
    TRADITION,
    MUSEUMS,
    NATURE,
    ART
}

export interface Question {
    id: number;
    questionType: QuestionType;
    text: string;
    answers: Answer[];

}